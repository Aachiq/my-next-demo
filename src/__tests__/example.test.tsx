import "@testing-library/jest-dom"; // to be included in each test file
import { render, screen } from "@testing-library/react";
import Page from "../app/page";

test("renders heading", () => {
  render(<Page />);
  expect(screen.getByRole("heading")).toBeInTheDocument();
});
test("check text if exist", () => {
  render(<Page />);
  expect(screen.getByText("Visit Channel")).toBeInTheDocument();
});

// snapshot tets next
// ðŸ†• Snapshot test
test("renders homepage unchanged", () => {
  const { container } = render(<Page />);
  expect(container).toMatchSnapshot();
});
