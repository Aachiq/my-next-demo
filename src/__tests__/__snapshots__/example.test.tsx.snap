// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders homepage unchanged 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-aajsq"
    >
      <div
        class="MuiBox-root css-0"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1fz1gr1-MuiTypography-root"
        >
          Home
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 css-zsmq4j-MuiTypography-root"
        >
          this is description
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary css-1ioaic5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Visit Channel
        </button>
      </div>
    </div>
  </div>
</div>
`;
